accepted/2024/NuGet-Audit-for-PackageDownload.md

```diff
@@ -0,0 +1,123 @@
+# ***Enhancing NuGet Audit for Secure and Selective `<PackageDownload>` Vulnerability Reporting***
+<!-- Replace `Title` with an appropriate title for your design -->
+
+- Author Name [Nigusu](https://github.com/Nigusu-Allehu)
+- [GitHub Issue](https://github.com/NuGet/Home/issues/13658)
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+To improve the security of NuGet, we are expanding NuGet Audit's support to include `<PackageDownload>` packages, allowing it to report vulnerabilities for both user-specified and SDK-managed packages.
+To prevent user confusion about SDK-included dependencies, we are introducing a new attribute, `AddedBy`, to identify the origin of each `<PackageDownload>`.
+This will provide users with clear, actionable security information while helping them distinguish between SDK-provisioned packages and those they added manually.
+
+## Motivation
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+
+We are expanding NuGet Audit to `<PackageDownload>` packages to improve security notifications, addressing gaps in vulnerability alerts that currently miss these packages.
+This change prevents confusion by clarifying SDK-managed PackageDownloads in the warning reports, ensuring users have an actionable warning.
+The expected outcome is a more precise and useful vulnerability reporting experience, enabling users to address actual risks without confusing SDK-related alerts.
+
+## Explanation
+
+### Functional explanation
+
+The NuGet Audit now supports `<PackageDownload>` packages, allowing it to detect and alert users about vulnerabilities in any user-specified dependencies downloaded via `<PackageDownload>`. 
+This means you’ll now receive warnings if a package added via `<PackageDownload>` has known security issues, so you can quickly address them.
+
+To prevent confusion about SDK-managed packages, we introduce a new attribute, `AddedBy`, on `<PackageDownload>` entries.
+This attribute will specify the source of the package, with values like `"dotnetsdk"` for SDK-managed packages and `"user"` for user-added dependencies.
+When a package marked with `AddedBy="dotnetsdk"` has a vulnerability, the warning will indicate that the package was added by the SDK.
+This will help users understand that the package originates from the SDK and is not something they explicitly added.
+
+| Attribute | Value     | Warning   |
+|-----------|-----------|-----------|
+| AddedBy   | dotnetsdk | warning NU1906: Package 'System.Text.Json' 1.0.0, added by '.NET SDK', has a known vulnerability. Refer to the vulnerability details at https://github.com/advisories/GHSA-g3q9-xf95-8hp5 to assess any necessary actions. |
```

Comment:
I think NuGet should special-case AddedBy=".NET SDK", and output a message along the lines "You should install a newer version of the .NET SDK"

Reply:
I think special-casing `AddedBy=".NET SDK"` should be handled by the SDK. NuGet will log the warnings and add them to the assets file, leaving it to the SDK to decide how to act. This ensures clear responsibilities and avoids overlap.